apps/settings/js/panels/call/call_settings.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Because there are some APIs that are working like mutex, we have to 
+ * wrap them inside a special module and provide a simple interface for
+ * users to use.
+ *
+ * And also, we can use CallSettings.observe('isLocked', ...) to know
+ * what's the current status and do related UI stuffs (like blocking ...)
+ */
+define(function(require) {
+  'use strict';
+
+  var SettingsCache = require('modules/settings_cache');
+  var TaskScheduler = require('modules/task_scheduler');
+  var CallForwarding = require('panels/call/call_forwarding');
+
+  var CallSettings = function() {
+    this._cardIndex = 0;
+    this._conns = window.navigator.mozMobileConnections;
+    this._conn = this._conns[this._cardIndex];
+    this._taskScheduler = TaskScheduler();
+  };
+
+  CallSettings.prototype = {
+    get observe() {
+      // we will pass out the observe out
+      return this._taskScheduler.observe.bind(this._taskScheduler);
+    },
+
+    set cardIndex(cardIndex) {
```

Comment:
This interface design introduces an additional context on all other APIs, which implies that when we call to any of the APIs, we should be aware of the current card index.

As a singleton, it can be used by many panels and modules at the same time. Making the APIs relying on the context (_cardIndex in this case) causes them interfere with each other. The other problem would be before calling to the API we should check the current card index first. IMHO this seems error-prone.